File: Data Structures & Algorithms/max-water-container/submission-1.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int n = heights.size();

            vector<pair<int,int>> arr ; 
        int ans =0; 
        arr.push_back({heights[0],0});
        int arrSize = 0 ; 
        for(int i =1;i<n;i++){
            if (  heights[i] - arr[arrSize].first >=  i - arr[arrSize].second  ){
                arr.push_back({heights[i],i});
            }   
            for( const auto & [ x,y] : arr){
                ans = max( ans , (i-y)* min(x , heights[i]));
            }

        }
        return ans ; 
        
    }
};

File: Data Structures & Algorithms/count-paths/submission-1.java
class Solution {
    public int uniquePaths(int m, int n) {
        int [][] dp = new int [m+2][n+2];
        dp[0][1] = 1 ; 
        dp[1][0]=0;
        
           
        for(int i =0 ;i <m;i++){
           
            for(int  j =0 ; j <n;j++){
              
                dp[i+1][j+1] = dp[i+1][j] +dp[i][j+1];
            }
        }
        return dp[m][n];
        
    }
}
